client/src/components/Notification.jsx

```diff
@@ -1,10 +1,82 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import * as notifications from '../actions/notificationActionCreator.jsx';
+import Note from './NotificationEntry.jsx';
 
-const Notification = (props) => (
-  <div>
-    <p>Notification~~~</p>
-  </div>
-)
+class Notification extends React.Component {
+  constructor(props) {
+    super(props);
+    this.clickHandler =this.clickHandler.bind(this);
+  };
 
+  componentWillMount() {
+    const {dispatch, userId} = this.props;
+    if (userId) {
+      dispatch(notifications.fetchNotifications(true));
+      fetch(`/notifications/${userId}`)
+      .then(response => {
+        if(!response.ok) {
+          throw Error(response.statusText);
+        }
+        dispatch(notifications.fetchNotifications(false));
+        dispatch(notifications.fetchError(false));
+        return response.json();
+      })
+      .then(data => {
+        dispatch(notifications.fetchSuccess(data));
+      })
+      .catch(() => dispatch(notifications.fetchError(true)));
+    }
+  };
 
-export default Notification;
\ No newline at end of file
+  clickHandler(notificationId, index) {
+    const {dispatch, userId, notification} = this.props;
+    let notes = notification.notifications;
+
+    let newNotification = notes.slice(0, index).concat([{...notes[index], 'read': true}]).concat(notes.slice(index + 1));
```

Comment:
I think its because it is a notification object and i only wanted to change one property on that object. while creating a new notification object. if you have a better way of doing it please tell me.